### PR TITLE
Implement stats debug modus using contstants

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #14634

### Summary of Changes

implement debug modus using contstants cc @brianteeman It took longer that expected sorry for that :( see: https://github.com/joomla/joomla-cms/pull/14634
### Testing Instructions

- apply this patch
- confirm the GUI debug option is gone (Plugin settings advanced tab is gone)
- go to the stats file
- uncomment the line
- confirm the method `isDebugEnabled()` return true.

### Expected result

Works, without GUI option

### Actual result

Works, just with the GUI option

### Documentation Changes Required

None.

### CODE REVIEW

File: plugins/system/stats/stats.php [hunk 1]
```diff
@@ -9,6 +9,9 @@
 
 defined('_JEXEC') or die;
 
+// Uncomment the following line to enable debug mode (update notification email sent every single time)
```
Comment (brianteeman) [inline plugins/system/stats/stats.php]: Can we change this to

Uncomment the following line to enable debug mode for testing purposes. Note statistics will be sent on every page load.

Comment (zero-24) [inline plugins/system/stats/stats.php]: oops fixed. Thanks

Comment (brianteeman) [inline plugins/system/stats/stats.php]: sorry to be a pain - reading it again we need to change it again to 
Note: statistics..

Comment (zero-24) [inline plugins/system/stats/stats.php]: np. fixed.

**Comment by Quy** [comment]
Remove the following language strings:
PLG_SYSTEM_STATS_DEBUG_LABEL
PLG_SYSTEM_STATS_DEBUG_DESC

**Comment by brianteeman** [comment]
@quy our standard practice is not to remove strings as they might be used by third parties etc

**Comment by zero-24** [comment]
i have just added the deprecate comment to that strings 😄

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on fab068c564786f0baaccc131dde42d0e0cee73a0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15178">issues.joomla.org/tracker/joomla-cms/15178</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on fab068c564786f0baaccc131dde42d0e0cee73a0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15178">issues.joomla.org/tracker/joomla-cms/15178</a>.</sub>

**Comment by rjcf18** [comment]
I have tested this item :white_check_mark: successfully on fab068c564786f0baaccc131dde42d0e0cee73a0<br /><br />The GUI debug option is gone and the isDebugEnabled() method returns true.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15178">issues.joomla.org/tracker/joomla-cms/15178</a>.</sub>

**Comment by zero-24** [comment]
Thanks for the testers! RTC for 3.7.1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15178">issues.joomla.org/tracker/joomla-cms/15178</a>.</sub>

**Pull request merged**